pkg/config/tmpl/rulesconfig.go

```diff
@@ -1,73 +1,212 @@
 package tmpl
 
-import y "gopkg.in/yaml.v3"
+import (
+	"fmt"
+	"strconv"
 
-type RulesConfig struct {
-	Version int
-	Envs    []EnvConfig
+	y "gopkg.in/yaml.v3"
+)
+
+type RulesComponentType string
+
+const (
+	Output        RulesComponentType = "output"
+	StartSampling RulesComponentType = "startsampling"
+	Condition     RulesComponentType = "condition"
+	Sampler       RulesComponentType = "sampler"
+	Dropper       RulesComponentType = "dropper"
+)
+
+func RCTFromStyle(style string) RulesComponentType {
+	switch style {
+	case "startsampling":
+		return StartSampling
+	case "condition":
+		return Condition
+	case "sampler":
+		return Sampler
+	case "dropper":
+		return Dropper
```

Comment:
casting the const RulesComponentTypes to a string here would be nice to ensure this mapping is not broken by a rename.